src/utils/temp.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * Copyright (c) 2010-2014 Bruce Williams
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const os = require('os');
+const constants = require('constants');
+
+const RDWR_EXCL = constants.O_CREAT | constants.O_TRUNC | constants.O_RDWR | constants.O_EXCL;
+const tempDir = path.resolve(os.tmpdir());
+
+function generateName(suffix) {
+  const now = new Date();
+  return path.join(
+    tempDir,
+    [
+      'jscodeshift',
+      now.getFullYear(),
+      now.getMonth(),
+      now.getDate(),
+      '-',
+      process.pid,
+      '-',
+      (Math.random() * 0x100000000 + 1).toString(36),
+      suffix || '',
+    ].join('')
+  );
+}
+
+exports.open = function open(opts, callback) {
+  if (!opts) opts = {};
+  const target = generateName(opts.suffix);
+  return new Promise((resolve, reject) => {
```

Comment:
Can this use async/await with a promisified version of `fs.open` given the minimum Node version is v16 now?

Reply:
totally! I'll get that updated 👍 I wasn't sure about whether I should match existing patterns or not

Reply:
@Daniel15: That's moved over to `fs.promises`. I didn't refactor `Runner.js` fully as I didn't want to make this PR overstep what it's supposed to do.
I also caught some invalid code that wasn't cleaned up properly.

Reply:
I actually just noticed that the existing tests aren't covering this and haven't been failing despite the problematic code that I forgot to clean up, I'll have a look at what happened there